### PR TITLE
Get all the containers building and working again.

### PR DESCRIPTION
**Pull request by eric-anderson**
See sub-commits for all the changes.

### CODE REVIEW

**Review by HenryL27 — APPROVED** [review]
lgtm

File: apps/crawler/crawler/s3/crawler/s3_crawler.py [hunk 1]
```diff
@@ -19,6 +19,12 @@ def __init__(
         self._prefix = prefix
         self._file_storage_location = "./.data/.s3/downloads"
         self._s3_client = self._get_s3_client(boto_session_args, boto_session_kwargs)
+        if os.path.exists("/.dockerenv"):
+            s = os.stat("/app/.data/.s3")
+            if s.st_uid != 1000 or s.st_gid != 1000:
+                raise Exception(
```
Comment (alexaryn) [inline apps/crawler/crawler/s3/crawler/s3_crawler.py:25]: Better to raise RuntimeError.  Exception is meant only to be inherited from.

File: apps/jupyter/fixuser.py [hunk 2]
```diff
@@ -46,12 +46,15 @@ def fix_ids(uid, gid):
...
     print("WARNING: Fixing IDs. This step can take a long time", flush=True)
+    print("  Temporarily fixing /app to be owned by root to avoid usermod changing stuff")
+    subprocess.run(["chown", "root:root", "/app"])
     print("  Fixing app group to have gid", gid, flush=True)
     subprocess.run(["groupmod", "--gid", str(gid), "app"])
     print("  Fixing app user to have uid", uid, "and gid", gid, flush=True)
     subprocess.run(["usermod", "--uid", str(uid), "--gid", str(gid), "app"])
     print("  Running recursive chown", flush=True)
-    subprocess.run(["chown", "-R", "app:app", "/app"])
+    # crawl data shouldn't be changed to our fancy user
```
Comment (alexaryn) [inline apps/jupyter/fixuser.py:56]: So, crawl data stays root:root?

Comment (eric-anderson) [reply apps/jupyter/fixuser.py:56]: No, it stays 1000:1000
jupyter notebook can become 1005:1005

**Pull request merged**